plugins/manipulator.cpp

```diff
@@ -304,144 +316,372 @@ struct UnitInfo
             list_id_prof = 0;
         }
     } ids;
+
+    string notices;
+    int8_t color;
+    int age;
+    int arrival_group;
+    int active_index;
+
+    int work_aptitude = 0;
+    int skill_aptitude = 0;
+
+    int notice_score;
+    int demand=0;
+    int scholar=0;
+    int performer=0;
+    int martial=0;
+    int topskilled=0;
+
+    uint8_t column_aptitudes[NUM_LABORS];
+    int8_t column_hints[NUM_LABORS];
+    bool sort_grouping_hint;
 };
 
+int work_aptitude_avg = 0;
+int skill_aptitude_avg = 0;
+
 enum detail_cols {
     DETAIL_MODE_PROFESSION,
     DETAIL_MODE_SQUAD,
-    DETAIL_MODE_JOB
+    DETAIL_MODE_JOB,
+    DETAIL_MODE_NOTICE,
+    DETAIL_MODE_ATTRIBUTE,
+    DETAIL_MODE_MAX
 };
-enum altsort_mode {
-    ALTSORT_NAME,
-    ALTSORT_SELECTED,
-    ALTSORT_DETAIL,
-    ALTSORT_STRESS,
-    ALTSORT_ARRIVAL,
-    ALTSORT_MAX
+const char * const detailmode_shortnames[] = {
+  "Profession,  ",
```

Comment:
I guess this works, but it'd be better to just store the strings you need and add spaces/punctuation where you need them. (Maybe our stl_sprintf() would help, or some variation on OutputString(), not sure.)

Reply:
Having customs strings for displayed labels can ne more convienient to reconfigure than custom spacing logic.